pulsar-client-api/src/main/java/org/apache/pulsar/client/api/DeadLetterPolicy.java

```diff
@@ -49,4 +49,15 @@
      */
     private String deadLetterTopic;
 
+    /**
+     * Name of the initial subscription name of the dead letter topic.
+     * The default value is the subscription name of the consumer.
+     */
+    private String initSubscriptionName;
+
+    /**
+     * Name of the initial subscription type of the dead letter topic.
+     * The default value is `Exclusive`.
+     */
+    private SubscriptionType initSubscriptionType;
```

Comment:
I don't think we need to expose this setting. The consumer is immediately closed and then whoever is consumer will just automatically reset the subscription type.